packages/solid-virtual/src/index.tsx

```diff
@@ -55,22 +55,19 @@ function createVirtualizerBase<
   }
 
   const virtualizer = new Proxy(instance, handler)
-  virtualizer.setOptions(resolvedOptions)
 
   onMount(() => {
     const cleanup = virtualizer._didMount()
-    virtualizer._willUpdate()
     onCleanup(cleanup)
   })
 
-  createComputed(() => {
+  createEffect(() => {
     virtualizer.setOptions(
```

Comment:
To avoid potential infinite loops on reactivity, extract

```
const updatedOptions = mergeProps(resolvedOptions, options);
```

and then wrap remainder in `untrack`.